lib/msal-common/src/client/BrokerAuthorizationCodeClient.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthorizationCodeClient } from "./AuthorizationCodeClient";
+import { AuthorizationUrlRequest } from "../request/AuthorizationUrlRequest";
+import { RequestParameterBuilder } from "../request/RequestParameterBuilder";
+import { ScopeSet } from "../request/ScopeSet";
+import { AuthorizationCodeRequest } from "../request/AuthorizationCodeRequest";
+import { StringUtils } from "../utils/StringUtils";
+import { ClientAuthError } from "../error/ClientAuthError";
+import { ResponseHandler } from "../response/ResponseHandler";
+import { BrokerAuthenticationResult } from "../response/BrokerAuthenticationResult";
+import { GrantType } from "../utils/Constants";
+
+/**
+ * Oauth2.0 Authorization Code client implementing the broker protocol for browsers.
+ */
+export class BrokerAuthorizationCodeClient extends AuthorizationCodeClient {
```

Comment:
I would argue against this approach, as now we have to remember to add any new parameters into two classes. Could the existing client be augmented to easily support the need broker parameters instead?

Reply:
I agree. I can make some updates to make this more streamlined so we have less copied code.

Reply:
done.